Sources/PT.PM.Common/MessagePack/RootUstMessagePackSerializer.cs

```diff
@@ -1,22 +1,29 @@
-﻿using MessagePack.Formatters;
-using MessagePack;
 using System;
+using System.Collections;
 using System.Collections.Generic;
-using MessagePack.LZ4;
-using MessagePack.Resolvers;
+using System.Linq;
+using System.Reflection;
+using MessagePack;
+using PT.PM.Common.Exceptions;
 using PT.PM.Common.Files;
 using PT.PM.Common.Nodes;
+using PT.PM.Common.Reflection;
 
 namespace PT.PM.Common.MessagePack
 {
-    public class RootUstMessagePackSerializer : IFormatterResolver
+    public class RootUstMessagePackSerializer
     {
         [ThreadStatic]
         private static byte[] buffer;
 
-        private TextSpanFormatter textSpanFormatter;
-        private FileFormatter fileFormatter;
-        private RootUstFormatter rootUstFormatter;
+        private BinaryFile serializedFile;
+        private TextSpanSerializer textSpanSerializer;
+        private FileSerializer fileSerializer;
+
+        private readonly Stack<RootUst> rootAncestors = new Stack<RootUst>();
+        private readonly Stack<Ust> ancestors = new Stack<Ust>();
```

Comment:
It's unclear how `ancestors` and `rootAncestors` work, some comments would be very useful.

Reply:
Ok, I will add them.